app/apptesting/gamm.go

```diff
@@ -111,3 +111,21 @@ func (s *KeeperTestHelper) RunBasicSwap(poolId uint64) {
 	_, err = s.App.GAMMKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], poolId, msg.TokenIn, denoms[1], msg.TokenOutMinAmount)
 	s.Require().NoError(err)
 }
+
+func (s *KeeperTestHelper) RunBasicJoinPool(poolId uint64) {
+	denoms, err := s.App.GAMMKeeper.GetPoolDenoms(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	for _, denom := range denoms {
+		s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin(denom, sdk.NewInt(10000000))))
+	}
+
+	pool, err := s.App.GAMMKeeper.GetPoolAndPoke(s.Ctx, poolId)
+	s.Require().NoError(err)
+	totalPoolShare := pool.GetTotalShares()
+	totalPoolShare.Quo(sdk.NewInt(100000))
+
+	tokenIn, _, err := s.App.GAMMKeeper.JoinPoolNoSwap(s.Ctx, s.TestAccs[0], poolId, totalPoolShare.Quo(sdk.NewInt(100000)), sdk.Coins{})
```

Comment:
Why are we dividing the totalPoolShare by 100000 and then doing it again in the call to the keeper?

Reply:
that was from logging, deleted!